lib/Parse/ParseDecl.cpp

```diff
@@ -2451,8 +2451,10 @@ Parser::parseDecl(ParseDeclOptions Flags,
 
         ParserStatus Status;
         bool PreviousHadSemi = true;
+        SyntaxParsingContext IfConfigCtx(SyntaxContext, SyntaxKind::CodeBlockItemList);
         while (Tok.isNot(tok::pound_else, tok::pound_endif, tok::pound_elseif,
                          tok::eof)) {
+          SyntaxParsingContext IfConfigCtx(SyntaxContext, SyntaxKind::CodeBlockItem);
```

Comment:
Accepting arbitrary code block item *only* in `#if` block is weird.
Current `master` implementation has a problem where it produces `UnknownDecl` for member with trailing comma. e.g.:

```swift
class C {
  let x = 1;
}
```

A solution for this would be:
* Make `MemberDeclBlock.members` `CodeBlockItemList`; or
* Create `MemberDeclBlockItemList` and `MemberDeclBlockItem`(`Decl` with optional `;`)

@nkcsgexi What do you think?

Reply:
I changed the `#if` block to accept a `DeclList`.

As for the problem with declarations followed by `;` I would vote for the second option.

Reply:
The second option makes sense to me too.